From f2dbec7d1a3b9118680c2d29fe33fc054d0e0340 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Fri, 24 Oct 2014 11:12:37 +0800
Subject: [PATCH 06/10] css2400b0_v21: [WORKAROUND]replace param buffer
 enqueue/dequeue count assertion check by just warning
 log

BZ: 227438

This assertion check fails on low CPU freq stress test which leads to board
hang and reboot. For a better stability for HQ dogfood, replace it with
warning log and the issue is still being tracked by ICG BZ 4328.

Change-Id: Id0c99e2cada49ec23a71f68887401f9ee349868a
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-8527
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://android.intel.com:443/292699
---
 .../pci/atomisp2/css2400b0_v21/sh_css_params.c     |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_params.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_params.c
index 95024ec..a117dc0 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_params.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_params.c
@@ -4021,7 +4021,12 @@ sh_css_param_update_isp_params(struct ia_css_pipe *curr_pipe,
 			 * parameter sets and dequeued sets
 			 */
 			g_param_buffer_enqueue_count++;
-			assert(g_param_buffer_enqueue_count < g_param_buffer_dequeue_count+50);
+			/* WORKAROUND: replace assertion check by just warning log, the BZ
+			 * is tracked by ICG BZ 4328. */
+			if (g_param_buffer_enqueue_count >= g_param_buffer_dequeue_count+50)
+				IA_CSS_WARNING("param buffer enqueue count %d, dequeue count %d\n",
+			                       g_param_buffer_enqueue_count,
+			                       g_param_buffer_dequeue_count);
 
 			/*
 			 * Tell the SP which queues are not empty,
-- 
1.7.9.5

